cluster/certificates.go

```diff
@@ -0,0 +1,118 @@
+package cluster
+
+import (
+	"crypto/rsa"
+	"fmt"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/rancher/rke/hosts"
+	"github.com/rancher/rke/k8s"
+	"github.com/rancher/rke/pki"
+	"k8s.io/client-go/util/cert"
+)
+
+// SaveClusterCerts ...
+func SaveClusterCerts(crts map[string]pki.CertificatePKI) error {
+	logrus.Infof("[certificates] Save kubernetes certificates as secrets")
+	for crtName, crt := range crts {
+		err := saveCrtToKubernetes(crtName, crt)
+		if err != nil {
+			return fmt.Errorf("Failed to save certificate [%s] to kubernetes", crtName)
+		}
+	}
+	logrus.Infof("[certificates] Successfuly saved certificates as kubernetes secret [%s]", pki.CertificatesSecretName)
+	return nil
+}
+
+func GetClusterCerts(kubeConfigPath string) (map[string]pki.CertificatePKI, error) {
+	logrus.Infof("[certificates] Getting Cluster certificates from Kubernetes")
+	k8sClient, err := k8s.NewClient(pki.KubeAdminConfigPath)
```

Comment:
Are we creating a k8s client for each an every k8s command we issue? It shoudl be done just once, and shared later on